src/playbook.yaml

```diff
@@ -43,6 +72,14 @@
         owner: www-data
         group: www-data
         mode: 0755
+    
+    - name: Add per-instance config script to be run on each provisioned EC2 instance after this AMI is built
+      copy:
+        src: set-instance-id-config-files.sh
+        #  when copied to this path, `clould-init` will ensure that the script
+        #  will only run once on every EC2 instance that uses the AMI we are currently building
```

Comment:
the SC allows updating an instance.  do we know what happens in that scenario?  will it get run again?

Reply:
no, `cloud-init` has a file based semaphore for each instance-id (`/var/lib/cloud/instances/i-00000XYZ/sem/config_scripts_per_instance`) to avoid re-running the script. Only changing to a different instance-id would rerun the script

Reply:
i think an SC update might generate a new instance-id.  I'm not sure if it always does that or it only does that in specific situation.  we should probably double check if it's important.